Source/missiles.cpp

```diff
@@ -2804,7 +2732,9 @@ int AddMissile(Point src, Point dst, Direction midir, missile_id mitype, mienemy
 		PlaySfxLoc(missileData.mlSFX, missile.position.start);
 	}
 
+	pParentMissile = pParent;
 	missileData.mAddProc(missile, dst, midir);
+	pParentMissile = nullptr;

```

Comment:
I'm personally not a fan of introducing a global for what should be a function parameter. The `AddProc` signatures have changed a few times already so I think it'd be fine to add a new param back in. Could add an `if (missileData.mAddProc == AddMisexp)` or similar check and pulling out the logic which refers to `parent` if you really wanted to avoid changing the function signatures for all the `AddProc`s, wouldn't be my preferred approach though.

Reply:
I agree, but just didn't have enough motivation to update all `Add` Methods.
If you want I can go the extra mile.

Reply:
I added a new `AddMissileParameter` struct that passes the parent to all `Add` Methods.
Now I'm tired. 😩